src/HomeTask5/Task2/MainBank.java

```diff
@@ -0,0 +1,29 @@
+package HomeTask5.Task2;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+
+public class MainBank {
+
+    public static void main(String[] args) {
+        Set<Account> accounts = new HashSet<>();
+        DebitAccount debitAccount = new DebitAccount("1", 500.0);
+        accounts.add(debitAccount);
+        accounts.add(new CreditAccount(2,200.0));
+
+        TreeSet<User> users = new TreeSet<>();
+        User userC = new User("Oksana");
+        users.add(userC);
+        users.add(new User("Olena"));
+        users.add(new User("Taras"));
+
+        UserServiceImpl userServiceImpl = new UserServiceImpl();
```

Comment:
Always go from generic type and set type of variable as interface

Reply:
Sorry but didn't get what exactly do you mean in the part of code
Could you please provide more details or example.

Reply:
thanks for reply in telegram
fixed according to advices UserService userService = new UserServiceImpl();